partner_communication_switzerland/data/sponsorship_communications_cron.xml

```diff
@@ -40,5 +31,88 @@
+            <field name="model_id" ref="model_recurring_contract"/>
+            <field name="config_id" ref="planned_anniversary_5"/>
+            <field name="state">communication</field>
+            <field name="partner_field">correspondent_id</field>
+            <field name="filter_domain">[("start_date", ">=", Datetime.now() - relativedelta(years=5))]</field>
+            <field name="trigger">on_time</field>
+            <field name="trg_date_id">start_date</field>
+            <field name="trg_date_range">60</field>
+            <field name="trg_date_range_type">month</field>
+            <field name="active" eval="False"/>
+        </record>
+        <record id="sponsorship_anniversary_10_communication" model="base.automation">
+            <field name="name">Send sponsorship anniversary 10 years communication</field>
+            <field name="model_id" ref="model_recurring_contract"/>
+            <field name="config_id" ref="planned_anniversary_10"/>
+            <field name="state">communication</field>
+            <field name="partner_field">correspondent_id</field>
+            <field name="filter_domain">[("start_date", ">=", Datetime.now() - relativedelta(years=10))]</field>
+            <field name="trigger">on_time</field>
+            <field name="trg_date_id">start_date</field>
+            <field name="trg_date_range">120</field>
+            <field name="trg_date_range_type">month</field>
+            <field name="active" eval="False"/>
+        </record>
+        <record id="sponsorship_anniversary_15_communication" model="base.automation">
+            <field name="name">Send sponsorship anniversary 15 years communication</field>
+            <field name="model_id" ref="model_recurring_contract"/>
+            <field name="config_id" ref="planned_anniversary_15"/>
+            <field name="state">communication</field>
+            <field name="partner_field">correspondent_id</field>
+            <field name="filter_domain">[("start_date", ">=", Datetime.now() - relativedelta(years=15))]</field>
+            <field name="trigger">on_time</field>
+            <field name="trg_date_id">start_date</field>
+            <field name="trg_date_range">180</field>
+            <field name="trg_date_range_type">month</field>
+            <field name="active" eval="False"/>
+        </record>
+
+        <!-- Write and pray reminders -->
+        <t t-set="min_date" t-value="Datetime.now() - relativedelta(months=3)"/>
+        <record id="wrpr_sponsorship_reminder" model="base.automation">
+            <field name="name">Write and Pray sponsorship reminder</field>
+            <field name="model_id" ref="model_recurring_contract"/>
+            <field name="config_id" ref="sponsorship_wrpr_reminder"/>
+            <field name="state">Send Communication</field>
+            <field name="partner_field">correspondent_id</field>
+            <field name="filter_domain">[("activation_date", ">=", min_date)]</field>
```

Comment:
You missed some filters that were used in the CRON. The problem is that the cron was checking if the sponsor wrote some letters with the field `sponsor_letter_ids` which is computed (therefore cannot be used in a search).

I propose that you add a field in object `recurring.contract` that will be useful for other purposes as well, that will capture the date at which the sponsor wrote his last letter. Please add the field in module `sbc_compassion` and update the date manually each time we create a new letter with direction `Supporter to Beneficiary`. 

```suggestion
            <field name="filter_domain">[("activation_date", ">=", min_date), ("state", "=", "active"), ("type", "=", "SC"), ("last_sponsor_letter", "=", False)]</field>
```